consensus/istanbul/utils.go

```diff
@@ -67,72 +67,56 @@ func CheckValidatorSignature(valSet ValidatorSet, data []byte, sig []byte) (comm
 	return common.Address{}, fmt.Errorf("not an elected validator %s", signer.Hex())
 }
 
-// Retrieves the block number within an epoch.  The return value will be 1-based.
-// There is a special case if the number == 0.  It is basically the last block of the 0th epoch, and should have a value of epochSize
+// GetNumberWithinEpoch retrieves the block number within an epoch.
+// The return value will be 1-based; thus first block of epoch is 1, and last block of epoch is `epochSize`
+// There is a special case if the number == 0. It is basically the last block of the 0th epoch, and should have a value of epochSize
 func GetNumberWithinEpoch(number uint64, epochSize uint64) uint64 {
 	number = number % epochSize
 	if number == 0 {
 		return epochSize
```

Comment:
This feels super-wrong, to say (even as a special case) that the genesis block is that epoch's 17280th block.  But I don't think it'd be easy to change this without making a bunch of changes elsewhere, so maybe we can let sleeping dogs lie on this one.

Reply:
agree it feels wrong, but i wouldn't dare changing this now